app/topic_info.py

```diff
@@ -0,0 +1,44 @@
+import json
+from typing import Any, Dict
+
+import pandas as pd
+
+
+class TopicInfo:
+    """Collect thresholding and quality information about one topic."""
+
+    def __init__(self, topic: str):
+        self.topic = topic
+        self.num_samples = 0
+        self.suggested_threshold = 0.5
+        self.scores = pd.DataFrame(
+            columns=['threshold', 'f1', 'precision', 'recall'])
+
+    def to_json_dict(self) -> Dict[str, Any]:
+        obj_dict = self.__dict__
+        obj_dict['scores'] = json.loads(self.scores.to_json(orient='index'))
+        return obj_dict
+
+    def load_json_dict(self, v: Dict[str, Any]) -> None:
+        self.__dict__ = v
+        self.scores = pd.DataFrame.from_dict(self.scores, orient='index')
+        self.scores.index = self.scores.index.map(float)
+
+    def get_quality(self) -> Any:
+        return self.scores.loc[self.suggested_threshold].f1
+
+    def get_confidence_at_probability(self, prob: float) -> Any:
+        # TODO: include distance between threshold and probability
+        scores = self.scores[self.scores.index >= prob]
+        if len(scores) > 0:
+            quality = scores.iloc[0].f1
```

Comment:
This is weird - by including the recall at the threshold (which f1 does), you're saying confidence_at_probability(1.0) is going to be much lower than 1?